### DISCUSSION TITLE
Unable to properly configure working version of podman.

### DISCUSSION THREAD

**Discussion by gabrielpantiru** [containers/podman#20780] [Q&A]
I am trying to use podman on my local machine however I cannot get it to work properly.

**First try with normal windows installer:**

After installing when trying to "Play pods from a KUbernetes YAML File" I get this error:
"Error: (HTTP code 500) server error - playing YAML file: initializing source docker://redis:latest: pinging container registry registry-1.docker.io: Get "https://registry-1.docker.io/v2/": dial tcp: lookup registry-1.docker.io: Temporary failure in name resolution"
 
For this I made sure to have the latest wsl (v2) and even tried to update /etc/resolv.conf file in it to 8.8.8.8 but did not worked.
The podman-config.yaml file from this tests: `apiVersion: v1
kind: Pod
metadata:
  name: my-pod41
spec:
  containers:
    - name: postgres
      image: postgres:latest
      env:
        - name: POSTGRES_PASSWORD
          value: POSTGRES_PASSWORD
        - name: POSTGRES_USER
          value: POSTGRES_USER
        - name: POSTGRES_DB
          value: POSTGRES_DB   
      ports:
        - containerPort: 5432
          hostPort: 5434
    - name: nats
      image: nats:2.9.14-alpine
`
![image](https://github.com/containers/podman/assets/117661610/97eea73f-6ee5-4a65-98a3-8609509da4f0)

**Second try with installer for restricted environments:**

there is a error saying that I need WSL version to be greater than 1.2.5 but lower it says that wsl2 is installed.

When I try to run the command wsl --version it displays the help part of the command, probably since it doesnt recognize the version part.

Can anyone help me with this? Do I need to have version 1 and 2 installed as well?

![image](https://github.com/containers/podman/assets/117661610/7c48b3d3-7295-4283-82dd-26dc7a32c1e3)

Obs: 

1. With the installer for restricted enviroments, I was able to pull remote images Eg: redis(Container Engine option did not appeared in this case). I was not able to run the pods because podman was not installed. 

3. With normal installer even when i try to pull a image the container engine appears Podman Machine but I get the error: "Error while pulling image from Podman Machine: access to image "redis" is denied (500 error). Can also be that the registry requires authentication."

![image](https://github.com/containers/podman/assets/117661610/f5b3e367-2471-4fdf-8f35-bd6fd61e1572)
If i try to pull the image by selecting docker as Container Engine instead of Podman Machine it is working properly
![image](https://github.com/containers/podman/assets/117661610/22f3572f-4721-47ce-89e5-cb388957da60)

I need to run any version that will allow me to run 3 containers from a yaml file ( redis, nats and postgres) to make a demo for podman as an alternative to docker. I have no vpn while making these tests. 

Anyone can offer some guidance ?

**Comment by gabrielpantiru** [comment]
Up, any ideas? 🤷‍♂️

**Comment by ghost** [comment]
any update ? the same as me

**Comment by rhatdan** [comment]
This looks like it is a podman desktop issue.

**Comment by rhatdan** [comment]
@ashley-cui @l0rd @n1hility PTAL

**Comment by l0rd** [comment]
@ljnszu @gabrielpantiru can you please provide more informations? What's your version of podman (`podman machine info` and `podman info` output)?  And what about the hosting OS (Windows right? What version/build?)?

And from what I understand there are 2 distinct problems:
1) Play kube fails because podman cannot connect to docker.io: can you try running `podman play kube podman-config.yaml` from a windows terminal and share the output?
2) Podman installation is not allowed because the wrong version of WSL is detected: can you try to download and install [podman-setup.exe](https://github.com/containers/podman/releases/download/v5.1.2/podman-5.1.2-setup.exe) and share the result?

**Reply by ghost** [reply]
Hi, I use windows 11 and wsl2 .I had installed docker for windows and used it correct before

Yesterday, I wanted to try podman and install podmanpodman-desktop-airgap-1.11.1-setup-x64.exe. It's ok to install podman-setup.exe and podman machine until I tryed to pulled the container image and displayed the message like that 

> Error while pulling image from Podman Machine: access to image "redis" is denied (500 error). Can also be that the registry requires authentication.

 I'm sorry that I didn't leave a screenshot. After that, I quited to try podman and turn back to use docker for windows. It's weird that my docker for windows started to reported error like this, not matter I uninstall podman and docker for window re-install.

> `starting services: initializing Docker API Proxy: setting up docker api proxy listener: open \\.\pipe\docker_engine: Access is denied.`

the final solution is kill the process proxy, I don't know who involved.
https://github.com/docker/for-win/issues/13663#issuecomment-1943460558

So the conclusion is this, it seems that there will be problems with the coexistence of docker for windows and podman and you can only choose one.
